src/app/test_executive/slot_end_test.ml

```diff
@@ -0,0 +1,200 @@
+open Core
+open Integration_test_lib
+
+module Make (Inputs : Intf.Test.Inputs_intf) = struct
+  open Inputs
+  open Engine
+  open Dsl
+
+  open Test_common.Make (Inputs)
+
+  (* TODO: find a way to avoid this type alias (first class module signatures restrictions make this tricky) *)
+  type network = Network.t
+
+  type node = Network.Node.t
+
+  type dsl = Dsl.t
+
+  let num_extra_keys = 100
+
+  let slot_tx_end = 5
+
+  let slot_chain_end = 8
+
+  let sender_account_prefix = "sender-account-"
+
+  let config =
+    let open Test_config in
+    { default with
+      requires_graphql = true
+    ; genesis_ledger =
+        (let open Test_account in
+        [ create ~account_name:"receiver-key" ~balance:"9999999" ()
+        ; create ~account_name:"sender-1-key" ~balance:"0" ()
+        ; create ~account_name:"sender-2-key" ~balance:"0" ()
+        ; create ~account_name:"sender-3-key" ~balance:"0" ()
+        ; create ~account_name:"snark-node-key" ~balance:"0" ()
+        ]
+        @ List.init num_extra_keys ~f:(fun i ->
+              create
+                ~account_name:(sprintf "%s-%d" sender_account_prefix i)
+                ~balance:"1000" () ))
+    ; block_producers =
+        [ { node_name = "receiver"; account_name = "receiver-key" }
+        ; { node_name = "sender-1"; account_name = "sender-1-key" }
+        ; { node_name = "sender-2"; account_name = "sender-2-key" }
+        ; { node_name = "sender-3"; account_name = "sender-3-key" }
+        ]
+    ; snark_coordinator =
+        Some
+          { node_name = "snark-node"
+          ; account_name = "snark-node-key"
+          ; worker_nodes = 4
+          }
+    ; txpool_max_size = 10_000_000
+    ; snark_worker_fee = "0.0002"
+    ; num_archive_nodes = 0
+    ; proof_config =
+        { proof_config_default with
+          work_delay = Some 1
+        ; transaction_capacity =
+            Some Runtime_config.Proof_keys.Transaction_capacity.small
+        }
+    ; slot_tx_end = Some slot_tx_end
+    ; slot_chain_end = Some slot_chain_end
+    }
+
+  let fee = Currency.Fee.of_nanomina_int_exn 10_000_000
+
+  let amount = Currency.Amount.of_nanomina_int_exn 10_000_000
+
+  let tx_delay_ms = 5000
+
+  let run network t =
+    let open Malleable_error.Let_syntax in
+    let logger = Logger.create () in
+    let num_slots = slot_chain_end + 2 in
```

Comment:
I made some changes to this test because it was either taking too much time or too flaky. I intend to backport these changes to `compatible`. Essentially, I reduced the overall number of slots needed to reach `slot_tx_end`, `slot_chain_end`, and the end of the test.